src/ziggurat/config.clj

```diff
@@ -108,3 +110,51 @@
 (defn -get [^String key]
   (let [config-vals (get config (keyword key))]
     (java-response config-vals)))
+
+;; for backward compatibility
+(def ^:private mapping-table
+  {:auto-offset-reset-config           :auto-offset-reset
+   :changelog-topic-replication-factor :replication-factor
+   :consumer-group-id                  :group-id
+   :default-api-timeout-ms-config      :default-api-timeout-ms
+   :key-deserializer-class-config      :key-deserializer
+   :key-serializer-class               :key-serializer
+   :session-timeout-ms-config          :session-timeout-ms
+   :stream-threads-count               :num-stream-threads
+   :value-deserializer-class-config    :value-deserializer
+   :value-serializer-class             :value-serializer})
+
+(def ^:private non-kafka-config-keys
+  [:consumer-type
+   :producer
+   :channels
+   :oldest-processed-message-in-s
+   :origin-topic
+   :input-topics
+   :join-cfg])
+
+(defn- to-string-key
+  [k]
+  (-> (get mapping-table k k)
+      (name)
+      (str/replace #"-" ".")))
+
+(defn- normalize-value
+  [v]
+  (cond
+    (or (boolean? v) (instance? Long v)) (str v)
+    (keyword? v)                         (name v)
+    :else                                v))
+
+(defn- set-property
+  [p k v]
+  (if (some #(= k %) non-kafka-config-keys)
+    p
+    (let [string-key (to-string-key k)
+          norm-value (normalize-value v)]
+      (.setProperty p string-key norm-value)
```

Comment:
we should do a trim keys and values and do a blank check on both before converting it to a string.

Reply:
clojure keywords cannot have spaces in them, no need to trim and check for blanks...

Reply:
We should at least do this check for values though?

Reply:
done